docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,520 @@
+| - | - |
+| `443` | ALPN port for TLS Routing. |
+
+</TabItem>
+<TabItem label="Separate Ports">
+
+These ports are required:
+
+| Port | Description |
+| - | - |
+| `3023` | SSH port for clients connect to. |
+| `3024` | SSH port used to create reverse SSH tunnels from behind-firewall environments. |
+| `443` | HTTPS connections to authenticate `tsh` users into the cluster. The same connection is used to serve a Web UI. |
+
+You can leave these ports closed if you are not using their corresponding
+services:
+
+| Port | Description |
+| - | - |
+| `3026` | HTTPS Kubernetes proxy |
+| `3036` | MySQL port |
+| `5432` | Postgres port |
+
+</TabItem>
+</Tabs>
+
+## Key-value store
+
+The Teleport Auth Service stores cluster state (such as dynamic configuration
+resources) and audit events as key/value pairs. In high-availability
+deployments, you must configure the Auth Service to manage this data in a
+key-value store that runs outside of your cluster of Teleport instances.
+
+The Auth Service supports the following backends for cluster state and audit
+events:
+
+- Amazon DynamoDB
+- Google Cloud Firestore
+
+For Amazon DynamoDB and Google Cloud Firestore, your Teleport configuration (which
+we will describe in more detail in the [Configuration](#configuration) section)
+names a table or collection where Teleport stores cluster state and audit
+events. 
+
+The Teleport Auth Service manages the creation of any required DynamoDB tables
+or Firestore collections itself, and does not require them to exist in advance.
+
+The Auth Service can also store cluster state in self-hosted
+[etcd](https://etcd.io/) deployments. In this case, Teleport uses namespaces
+within item keys to identify cluster state data.
+
+<Admonition title="Required permissions">
+
+In your cloud provider's RBAC solution (e.g., AWS or Google Cloud IAM), your
+Auth Service instances need permissions to read from and write to your chosen
+key/value store, as well as to create tables and collections (if your key/value
+store supports them).
+
+</Admonition>
+
+## Object storage service
+
+High-availability Teleport deployments use an object storage service for
+persisting session recordings. The Teleport Auth Service supports two object
+storage services:
+
+- Amazon S3 (or an S3-compatible object store)
+- Google Cloud Storage
+
+In your Teleport configuration (described in the [Configuration](#configuration)
+section), you must name a bucket within Google Cloud Storage or an Amazon
+S3-compatible service to use for managing session recordings. The Teleport Auth
+Service creates this bucket, so to prevent unexpected behavior, you should not
+create it in advance. 
+
+<Admonition title="Required permissions">
+
+In your cloud provider's RBAC solution, your Auth Service instances need
+permissions to create, read from, and write to buckets.
+
+</Admonition>
+
+## TLS credential provisioning 
+
+High-availability Teleport deployments require an automated system to fetch TLS
```

Comment:
Technically we don't require an _automated_ system. You can manually procure TLS certificates from any CA you'd like and still deploy Teleport in a highly available manner.

LetsEncrypt or ACM are certainly more convenient, but I'm not sure they impact the availability properties of the system (so long as you don't let your certs expire)

Reply:
Automatic cert-provisioning impacts the availability properties negatively because we poorly handle certificate reload. It requires breaking all existing sessions, + the Helm chart does not automatically reload certs, and Teleport never re-reads its files. Running with a good ol' DigiCert wildcard valid for 3 years provides better availability than using LE.

Reply:
👍